File: hw/tipantiza/u3/hw27ObserverPattern/ObserverPattern_py/model/ibm.py
from model.stock import Stock


class IBM(Stock):

    def __init__(self, symbol, price):
        super().__init__()
        self.symbol = symbol
        self.price = price

    def set_price(self, price):
        self.price = price
        self.notify_observers(price)

    def set_symbol(self, symbol):
        self.symbol = symbol
        self.notify_observers(symbol)

File: hw/codena/u3/hw26Observer/Java/Observer/src/main/java/ec/edu/espe/observer/model/IBM.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.observer.model;

/**
 *
 * @author Daniel Codena, CodeBreakers, @ESPE
 */
public class IBM extends Stock {

    public IBM(String symbol, double price) {
        this.symbol = symbol;
        this.price = price;

    }

    @Override
    public void setPrice(double price) {
        this.price = price;
        notifyObservers(price);
    }

    @Override
    public void setSymbol(String symbol) {
        this.symbol = symbol;
        notifyObservers(symbol);
    }

}
